snap/snapcraft.yaml

```diff
@@ -0,0 +1,64 @@
+name: charmcraft
+base: core20
+summary: The charming tool
+license: Apache-2.0
+description: |
+  Charmcraft provides a streamlined, powerful, opinionated, and
+  flexible tool to develop, package, and manage the lifecycle of Juju
+  charm publication, focused particularly on charms written within the
+  Operator Framework.
+
+  It is still in heavy, initial development and so a lot is still To
+  Be Done. However it is already useful, and already simplifies the
+  life of the charmer.
+
+adopt-info: charmcraft  # look for 'snapcraftctl set-*' in the charmcraft part
+architectures: [all]    # nothing platform-dependent here (for now at least)
+
+apps:
+  charmcraft:
+    command: bin/charmcraft
+    plugs:
+      - home            # so it can acess files under the user's home
+      - removable-media # so it can access things in /media, etc. (manually connected)
+      - network         # so 'pip install' can download things
```

Comment:
Do any of these need official approval? Or are they always available to any charm since they are likely needed by most?

Reply:
These in particular don't. When we pull in git and/or juju we'd need to ask for read permission on the config files and such, which is a reasonably straightforward process (I did it for `icdiff` for example).